samples/wifi/raw_tx_packet/src/main.c

```diff
@@ -0,0 +1,515 @@
+
+	mode_info.mode =  WIFI_STA_MODE | WIFI_TX_INJECTION_MODE;
+
+	ret = net_mgmt(NET_REQUEST_WIFI_MODE, iface, &mode_info, sizeof(mode_info));
+	if (ret) {
+		LOG_ERR("Mode setting failed %d", ret);
+	}
+}
+
+static int setup_raw_pkt_socket(int *sockfd, struct sockaddr_ll *sa)
+{
+	struct net_if *iface;
+	int ret;
+
+	*sockfd = socket(AF_PACKET, SOCK_RAW, IPPROTO_RAW);
+	if (*sockfd < 0) {
+		LOG_ERR("Unable to create a socket %d", errno);
+		return -1;
+	}
+
+	iface = net_if_get_default();
+	if (!iface) {
+		LOG_ERR("Unable to get default interface");
+		return -1;
+	}
+
+	sa->sll_family = AF_PACKET;
+	sa->sll_ifindex = net_if_get_by_iface(iface);
+
+	/* Bind the socket */
+	ret = bind(*sockfd, (struct sockaddr *)sa, sizeof(struct sockaddr_ll));
+	if (ret < 0) {
+		LOG_ERR("Error: Unable to bind socket to the network interface:%d", errno);
+		close(*sockfd);
+		return -1;
+	}
+
+	return 0;
+}
+
+static void fill_raw_tx_pkt_hdr(struct nrf_wifi_fmac_rawpkt_info *raw_tx_pkt)
+{
+	/* Raw Tx Packet header */
+	raw_tx_pkt->magic_number = 0x12345678;
+	raw_tx_pkt->data_rate = CONFIG_RAW_TX_PACKET_SAMPLE_RATE_VALUE;
+	raw_tx_pkt->packet_length = sizeof(test_beacon_frame);
+	raw_tx_pkt->tx_mode = CONFIG_RAW_TX_PACKET_SAMPLE_RATE_FLAGS;
+	raw_tx_pkt->queue = CONFIG_RAW_TX_PACKET_SAMPLE_QUEUE_NUM;
+	raw_tx_pkt->reserved = 0;
+}
+
+static int print_pkt_count_info(int *num_tx_pkts)
+{
+#if defined(CONFIG_RAW_TX_PACKET_SAMPLE_FIXED_NUM_PACKETS)
+	*num_tx_pkts = CONFIG_RAW_TX_PACKET_SAMPLE_FIXED_NUM_PACKETS;
+	if (*num_tx_pkts == 0) {
+		LOG_ERR("Can't send %d number of raw tx packets", *num_tx_pkts);
+		return -1;
+	}
+	LOG_INF("Sending %d number of raw tx packets", *num_tx_pkts);
+#else
+	*num_tx_pkts = -1;
+	LOG_INF("Sending raw tx packets continuously");
+#endif
+	return 0;
+}
+
+static void wifi_send_raw_tx_packet(void)
+{
+	struct sockaddr_ll sa;
+	int sockfd, ret;
+	struct nrf_wifi_fmac_rawpkt_info packet;
+	char *test_frame = NULL;
+	int buf_length, num_pkts, transmission_mode;
+
+	ret = setup_raw_pkt_socket(&sockfd, &sa);
+	if (ret < 0) {
+		LOG_ERR("Setting socket for raw pkt transmission failed %d", errno);
+		return;
+	}
+
+	fill_raw_tx_pkt_hdr(&packet);
+
+	ret = print_pkt_count_info(&num_pkts);
+	if (ret < 0) {
+		close(sockfd);
+		return;
+	}
+
+	test_frame = malloc(sizeof(struct nrf_wifi_fmac_rawpkt_info) + sizeof(test_beacon_frame));
+	if (!test_frame) {
+		LOG_ERR("Malloc failed for send buffer %d", errno);
+		return;
+	}
+
+	buf_length = sizeof(struct nrf_wifi_fmac_rawpkt_info) + sizeof(test_beacon_frame);
+	memcpy(test_frame, &packet, sizeof(struct nrf_wifi_fmac_rawpkt_info));
+	memcpy(test_frame + sizeof(struct nrf_wifi_fmac_rawpkt_info),
+				&test_beacon_frame, sizeof(test_beacon_frame));
+
+	transmission_mode = CONFIG_RAW_TX_PACKET_SAMPLE_PACKET_TRANSMISSION_MODE;
+	while (1) {
+		if (transmission_mode == FIXED_MODE_TRANSMISSION && num_pkts == 0) {
+			break;
+		}
+
+		ret = sendto(sockfd, test_frame, buf_length, 0,
```

Comment:
Sending same packet makes it tough to debug, can you please increase `seq_ctrl` per-packet(make sure to handle wrap around at 4095)?